Content.Server/Silicons/Borgs/BorgSystem.Transponder.cs

```diff
@@ -0,0 +1,107 @@
+using Content.Shared.Emag.Components;
+using Content.Shared.Popups;
+using Content.Shared.Research;
+using Content.Shared.Silicons.Borgs.Components;
+using Content.Server.DeviceNetwork;
+using Content.Server.DeviceNetwork.Components;
+using Content.Server.DeviceNetwork.Systems;
+using Content.Server.Explosion.Components;
+
+namespace Content.Server.Silicons.Borgs;
+
+/// <inheritdoc/>
+public sealed partial class BorgSystem
+{
+    private void InitializeTransponder()
+    {
+        SubscribeLocalEvent<BorgTransponderComponent, DeviceNetworkPacketEvent>(OnPacketReceived);
+    }
+
+    public override void Update(float frameTime)
+    {
+        base.Update(frameTime);
+
+        var now = _timing.CurTime;
+        var query = EntityQueryEnumerator<BorgTransponderComponent, BorgChassisComponent, DeviceNetworkComponent, MetaDataComponent>();
+        while (query.MoveNext(out var uid, out var comp, out var chassis, out var device, out var meta))
+        {
+            if (now < comp.NextBroadcast)
+                continue;
+
+            var charge = 0f;
+            if (_powerCell.TryGetBatteryFromSlot(uid, out var battery))
+                charge = battery.Charge / battery.MaxCharge;
+
+            var data = new CyborgControlData()
+            {
+                Chassis = meta.EntityPrototype!.ID,
```

Comment:
Can't null suppress entity prototypes as there's no guarantee they exist + you shouldn't be relying upon IDs for this.

Reply:
how should i be giving the original prototype name for model (or just not have it) and the icon

Reply:
Borg component or whatever, just not the entityprototype.

Reply:
can do it for the name but what about icon

Reply:
chuck it somewhere else that doesn't have a chance of nullreffing.